mkosi/__init__.py

```diff
@@ -5308,8 +5405,10 @@ def xescape(s: str) -> str:
     return ret
 
 
-def build_auxiliary_output_path(path: str, suffix: str, xz: Optional[bool] = False) -> str:
-    return strip_suffixes(path) + suffix + (".xz" if xz else "")
+def build_auxiliary_output_path(args: argparse.Namespace, suffix: str, can_compress: bool = False) -> str:
+    output = strip_suffixes(args.output)
+    compress = should_compress_output(args) if can_compress else False
+    return output + suffix + (f".{compress}" if compress else "")
```

Comment:
Just a stylistic quibble, feel free to ignore it
```suggestion
    return f"{output}{suffix}.{compress or ''}"
```

Reply:
No, that makes the dot unconditional, but it should be added only if the compression suffix is added.

Reply:
Right, chalk it up to post-vaccine grogginess :)